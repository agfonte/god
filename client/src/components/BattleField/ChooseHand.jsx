import React from "react";
import { Row, Button, Col, Container } from "react-bootstrap";
export default function ChooseHand(props) {
  let { moves } = props;
  const chooseHand = (
    <Container>
      <Row id="test-array-moves" className={"justify-content-center text-center"}>
        {moves.map(mov => {
          return (
            <Col key={mov} className={"text-center"}>
              <Button
                id={"test-move-" + { mov }}
                onClick={evt => {
                  props.onChoose(mov, props.user);
                }}
              >
                {mov.toUpperCase()}
              </Button>
            </Col>
          );
        })}
      </Row>
    </Container>
  );
  return chooseHand;
}
