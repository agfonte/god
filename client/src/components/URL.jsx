export const urls = {
  users: "http://localhost:4000/api/users",
  moves: "http://localhost:4000/api/moves",
  settings: "http://localhost:4000/api/settings"
};
