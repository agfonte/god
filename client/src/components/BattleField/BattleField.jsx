import React, { Component } from "react";
import PropTypes from "prop-types";
import { Row, Container, Col } from "react-bootstrap";
import "bootstrap/dist/css/bootstrap.min.css";
import ChooseHand from "./ChooseHand";
import ShowHandWinner from "./ShowHandWinner";
import ShowChampion from "./ShowChampion";
import { urls } from "../URL";
class BattleField extends Component {
  state = {
    currentPlayer: this.props.user1,
    winner: undefined,
    user1: this.props.user1,
    user2: this.props.user2,
    round: 1,
    stats: {
      user1: 0,
      user2: 0
    },
    resolve: false,
    finished: false,
    currentHandP1: undefined,
    currentHandP2: undefined,
    moves: [],
    kills: {},
    style: "red"
  };
  componentDidMount() {
    const axios = require("axios");
    axios
      .get(urls.moves)
      .then(resmoves => {
        let m = resmoves.data.moves;
        let set = new Set();
        m.forEach(move => {
          set.add(move.move[0]);
          set.add(move.move[1]);
        });
        this.setState({ kills: resmoves.data.moves, moves: [...set] });
      })
      .catch(err => {
        return <div />;
      });
  }
  onChoose = (mov, user) => {
    let { user1, user2 } = this.props;
    if (user === user1) {
      this.setState({
        currentHandP1: mov,
        currentPlayer: user2,
        style: "green"
      });
    } else {
      this.setState(
        {
          currentHandP2: mov,
          currentPlayer: user1,
          resolve: true,
          style: "red"
        },
        () => {
          this.resolveBattle();
        }
      );
    }
  };
  resolveBattle = () => {
    let { stats, currentHandP1, currentHandP2, user1, user2 } = this.state;
    const win = this.resolveHandWin(currentHandP1, currentHandP2);
    let winner;
    if (win === 0) {
      stats.user1++;
      winner = user1;
      this.props.onRoundWin(user1);
    } else if (win === 1) {
      stats.user2++;
      winner = user2;
      this.props.onRoundWin(user2);
    } else {
      winner = undefined;
    }
    if (stats.user1 === 3 || stats.user2 === 3) {
      this.setState(
        {
          winner
        },
        () => {
          this.endBattle();
        }
      );
      return;
    } else {
      this.setState({
        stats: stats,
        winner: winner
      });
    }
  };
  resolveHandWin = (currentHandP1, currentHandP2) => {
    let { kills } = this.state;

    for (let move in kills) {
      if (
        kills[move].move[0] === currentHandP1 &&
        kills[move].move[1] === currentHandP2
      ) {
        return 0;
      } else {
        if (
          kills[move].move[0] === currentHandP2 &&
          kills[move].move[1] === currentHandP1
        ) {
          return 1;
        }
      }
    }
    return 2;
  };
  endBattle = () => {
    const axios = require("axios");
    let { stats, user1, user2 } = this.state;
    axios
      .put(urls.users, {
        name: user1,
        win: stats.user1,
        lose: stats.user2,
        against: user2
      })
      .then(res => {
        // console.log(res);
      })
      .catch(err => {
        // console.log(err);
      });
    axios
      .put(urls.users, {
        name: user2,
        win: stats.user2,
        lose: stats.user1,
        against: user1
      })
      .then(res => {
        // console.log(res);
      })
      .catch(err => {
        // console.log(err);
      });
    this.setState({
      stats: { user1: 0, user2: 0 },
      resolve: false,
      finished: true,
      currentHandP1: undefined,
      currentHandP2: undefined,
      round: 1
    });
    this.props.onBattleEnded();
  };
  nextRound = () => {
    let round = this.state.round;
    this.setState({
      round: round + 1,
      winner: undefined,
      resolve: false,
      currentHandP1: undefined,
      currentHandP2: undefined
    });
  };
  playAgain = () => {
    this.setState({
      stats: { user1: 0, user2: 0 },
      resolve: false,
      finished: false,
      currentHandP1: undefined,
      currentHandP2: undefined,
      winner: undefined
    });
  };
  render() {
    let { round, currentPlayer, moves, finished } = this.state;
    let { user1, user2 } = this.props;
    return (
      <Container>
        <Row>
          <Col>
            <input
              type={"image"}
              id={"exit-img"}
              src={"./back.png"}
              alt={"Home icon"}
              style={{
                height: "40px",
                width: "40px",
                background: "transparent"
              }}
              onClick={this.props.back}
            />
          </Col>
        </Row>
        <Row className={"justify-content-center"}>
          <h1 style={{ color: "white", fontSize: "4rem" }}>Round {round}</h1>
        </Row>
        <Row className={"justify-content-center"}>
          <h2 style={{ color: "white" }}>
            <Row className={"justify-content-center"}>
              <p style={{ color: "red" }}>{user1}</p>
            </Row>
            <Row className={"justify-content-center"}>
              <strong>
                <i>vs</i>
              </strong>
            </Row>
            <Row className={"justify-content-center"}>
              <p style={{ color: "green" }}>{user2}</p>
            </Row>
          </h2>
        </Row>
        <Row className={"justify-content-center"}>
          <h3 style={{ color: "white" }} className={"mr-2"}>
            Choose your hand{" "}
          </h3>
          <h3 style={{ color: this.state.style }}>
            {this.state.currentPlayer}
          </h3>
        </Row>
        <Row>
          <ChooseHand
            user={currentPlayer}
            round={round}
            moves={moves}
            onChoose={this.onChoose}
          />
        </Row>
        <ShowHandWinner
          winner={this.state.winner}
          round={round}
          handUser1={this.state.currentHandP1}
          handUser2={this.state.currentHandP2}
          show={this.state.resolve}
          handleCloseModal={this.nextRound}
        />
        <ShowChampion
          playAgain={this.playAgain}
          show={finished}
          winner={this.state.winner}
          homeScreen={this.props.back}
        />
        ;
      </Container>
    );
  }
}

BattleField.propTypes = {
  user1: PropTypes.string,
  user2: PropTypes.string,
  back: PropTypes.func
};
BattleField.defaultProps = {
  user1: "",
  user2: "",
  back: () => {}
};
export default BattleField;
