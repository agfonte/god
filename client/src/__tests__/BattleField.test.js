import React from "react";
import BattleField from "../components/BattleField/BattleField";
import { urls } from "../components/URL";
import { shallow, mount } from "enzyme";
const axios = require("axios");
describe("<BattleField />", () => {
  it("should be define Game", () => {
    expect(BattleField).toBeDefined();
  });
  it("renders 1 <BattleField /> component", () => {
    const component = shallow(<BattleField />);
    expect(component).toHaveLength(1);
  });

  it("should have right amount of moves", async () => {
    let moves = await axios.get(urls.moves);
    moves = moves.data.moves;
    let set = new Set();
    moves.forEach(move => {
      set.add(move.move[0]);
      set.add(move.move[1]);
    });

    const component = mount(<BattleField user1="user1" user2="user2" />);
    const choose = component.find("#test-array-moves");
    expect(choose).toHaveLength(set.size);
  });
  
    it("renders 1 <Game /> component", () => {
      const component = shallow(<BattleField />);
      expect(component).toMatchSnapshot();
    });
  // it("Should resolve round win", async () => {
  //   let moves = await axios.get(urls.moves);
  //   moves = moves.data.moves;
  //   const mockWinner = jest.fn(wuser => {
  //     console.log("joasdaosdk");
  //     return wuser;
  //   });
  //   const component = shallow(
  //     <BattleField
  //       user1="user1"
  //       user2="user2"
  //       currentHandP1={moves[0].move[0]}
  //       currentHandP2={moves[0].move[0]}
  //       onRoundWin={mockWinner}
  //     />
  //   );
  //   const resolve = jest.spyOn(component.instance(), "resolveBattle");
  //   console.log(resolve());
  //   console.log(mockWinner.mock.calls);
  // });
});
