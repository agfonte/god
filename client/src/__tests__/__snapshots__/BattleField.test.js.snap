// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BattleField /> renders 1 <Game /> component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BattleField
    back={[Function]}
    user1=""
    user2=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef(Bootstrap(Row))>
          <ForwardRef(Bootstrap(Col))>
            <input
              alt="Home icon"
              id="exit-img"
              onClick={[Function]}
              src="./back.png"
              style={
                Object {
                  "background": "transparent",
                  "height": "40px",
                  "width": "40px",
                }
              }
              type="image"
            />
          </ForwardRef(Bootstrap(Col))>
        </ForwardRef(Bootstrap(Row))>,
        <ForwardRef(Bootstrap(Row))
          className="justify-content-center"
        >
          <h1
            style={
              Object {
                "color": "white",
                "fontSize": "4rem",
              }
            }
          >
            Round 
            1
          </h1>
        </ForwardRef(Bootstrap(Row))>,
        <ForwardRef(Bootstrap(Row))
          className="justify-content-center"
        >
          <h2
            style={
              Object {
                "color": "white",
              }
            }
          >
            <ForwardRef(Bootstrap(Row))
              className="justify-content-center"
            >
              <p
                style={
                  Object {
                    "color": "red",
                  }
                }
              >
                
              </p>
            </ForwardRef(Bootstrap(Row))>
            <ForwardRef(Bootstrap(Row))
              className="justify-content-center"
            >
              <strong>
                <i>
                  vs
                </i>
              </strong>
            </ForwardRef(Bootstrap(Row))>
            <ForwardRef(Bootstrap(Row))
              className="justify-content-center"
            >
              <p
                style={
                  Object {
                    "color": "green",
                  }
                }
              >
                
              </p>
            </ForwardRef(Bootstrap(Row))>
          </h2>
        </ForwardRef(Bootstrap(Row))>,
        <ForwardRef(Bootstrap(Row))
          className="justify-content-center"
        >
          <h3
            className="mr-2"
            style={
              Object {
                "color": "white",
              }
            }
          >
            Choose your hand
             
          </h3>
          <h3
            style={
              Object {
                "color": "red",
              }
            }
          >
            
          </h3>
        </ForwardRef(Bootstrap(Row))>,
        <ForwardRef(Bootstrap(Row))>
          <ChooseHand
            moves={Array []}
            onChoose={[Function]}
            round={1}
            user=""
          />
        </ForwardRef(Bootstrap(Row))>,
        <ShowHandWinner
          handleCloseModal={[Function]}
          round={1}
          show={false}
        />,
        <ShowChampion
          homeScreen={[Function]}
          playAgain={[Function]}
          show={false}
        />,
        ";",
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef(Bootstrap(Col))>
            <input
              alt="Home icon"
              id="exit-img"
              onClick={[Function]}
              src="./back.png"
              style={
                Object {
                  "background": "transparent",
                  "height": "40px",
                  "width": "40px",
                }
              }
              type="image"
            />
          </ForwardRef(Bootstrap(Col))>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <input
              alt="Home icon"
              id="exit-img"
              onClick={[Function]}
              src="./back.png"
              style={
                Object {
                  "background": "transparent",
                  "height": "40px",
                  "width": "40px",
                }
              }
              type="image"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Home icon",
              "id": "exit-img",
              "onClick": [Function],
              "src": "./back.png",
              "style": Object {
                "background": "transparent",
                "height": "40px",
                "width": "40px",
              },
              "type": "image",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "propTypes": undefined,
            "render": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <h1
            style={
              Object {
                "color": "white",
                "fontSize": "4rem",
              }
            }
          >
            Round 
            1
          </h1>,
          "className": "justify-content-center",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Round ",
              1,
            ],
            "style": Object {
              "color": "white",
              "fontSize": "4rem",
            },
          },
          "ref": null,
          "rendered": Array [
            "Round ",
            1,
          ],
          "type": "h1",
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <h2
            style={
              Object {
                "color": "white",
              }
            }
          >
            <ForwardRef(Bootstrap(Row))
              className="justify-content-center"
            >
              <p
                style={
                  Object {
                    "color": "red",
                  }
                }
              >
                
              </p>
            </ForwardRef(Bootstrap(Row))>
            <ForwardRef(Bootstrap(Row))
              className="justify-content-center"
            >
              <strong>
                <i>
                  vs
                </i>
              </strong>
            </ForwardRef(Bootstrap(Row))>
            <ForwardRef(Bootstrap(Row))
              className="justify-content-center"
            >
              <p
                style={
                  Object {
                    "color": "green",
                  }
                }
              >
                
              </p>
            </ForwardRef(Bootstrap(Row))>
          </h2>,
          "className": "justify-content-center",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ForwardRef(Bootstrap(Row))
                className="justify-content-center"
              >
                <p
                  style={
                    Object {
                      "color": "red",
                    }
                  }
                >
                  
                </p>
              </ForwardRef(Bootstrap(Row))>,
              <ForwardRef(Bootstrap(Row))
                className="justify-content-center"
              >
                <strong>
                  <i>
                    vs
                  </i>
                </strong>
              </ForwardRef(Bootstrap(Row))>,
              <ForwardRef(Bootstrap(Row))
                className="justify-content-center"
              >
                <p
                  style={
                    Object {
                      "color": "green",
                    }
                  }
                >
                  
                </p>
              </ForwardRef(Bootstrap(Row))>,
            ],
            "style": Object {
              "color": "white",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <p
                  style={
                    Object {
                      "color": "red",
                    }
                  }
                >
                  
                </p>,
                "className": "justify-content-center",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                  "style": Object {
                    "color": "red",
                  },
                },
                "ref": null,
                "rendered": "",
                "type": "p",
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": undefined,
                "propTypes": undefined,
                "render": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <strong>
                  <i>
                    vs
                  </i>
                </strong>,
                "className": "justify-content-center",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <i>
                    vs
                  </i>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "vs",
                  },
                  "ref": null,
                  "rendered": "vs",
                  "type": "i",
                },
                "type": "strong",
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": undefined,
                "propTypes": undefined,
                "render": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <p
                  style={
                    Object {
                      "color": "green",
                    }
                  }
                >
                  
                </p>,
                "className": "justify-content-center",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                  "style": Object {
                    "color": "green",
                  },
                },
                "ref": null,
                "rendered": "",
                "type": "p",
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": undefined,
                "propTypes": undefined,
                "render": [Function],
              },
            },
          ],
          "type": "h2",
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <h3
              className="mr-2"
              style={
                Object {
                  "color": "white",
                }
              }
            >
              Choose your hand
               
            </h3>,
            <h3
              style={
                Object {
                  "color": "red",
                }
              }
            >
              
            </h3>,
          ],
          "className": "justify-content-center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Choose your hand",
                " ",
              ],
              "className": "mr-2",
              "style": Object {
                "color": "white",
              },
            },
            "ref": null,
            "rendered": Array [
              "Choose your hand",
              " ",
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "style": Object {
                "color": "red",
              },
            },
            "ref": null,
            "rendered": "",
            "type": "h3",
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ChooseHand
            moves={Array []}
            onChoose={[Function]}
            round={1}
            user=""
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "moves": Array [],
            "onChoose": [Function],
            "round": 1,
            "user": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handUser1": undefined,
          "handUser2": undefined,
          "handleCloseModal": [Function],
          "round": 1,
          "show": false,
          "winner": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "homeScreen": [Function],
          "playAgain": [Function],
          "show": false,
          "winner": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      ";",
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "defaultProps": undefined,
      "propTypes": undefined,
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef(Bootstrap(Row))>
            <ForwardRef(Bootstrap(Col))>
              <input
                alt="Home icon"
                id="exit-img"
                onClick={[Function]}
                src="./back.png"
                style={
                  Object {
                    "background": "transparent",
                    "height": "40px",
                    "width": "40px",
                  }
                }
                type="image"
              />
            </ForwardRef(Bootstrap(Col))>
          </ForwardRef(Bootstrap(Row))>,
          <ForwardRef(Bootstrap(Row))
            className="justify-content-center"
          >
            <h1
              style={
                Object {
                  "color": "white",
                  "fontSize": "4rem",
                }
              }
            >
              Round 
              1
            </h1>
          </ForwardRef(Bootstrap(Row))>,
          <ForwardRef(Bootstrap(Row))
            className="justify-content-center"
          >
            <h2
              style={
                Object {
                  "color": "white",
                }
              }
            >
              <ForwardRef(Bootstrap(Row))
                className="justify-content-center"
              >
                <p
                  style={
                    Object {
                      "color": "red",
                    }
                  }
                >
                  
                </p>
              </ForwardRef(Bootstrap(Row))>
              <ForwardRef(Bootstrap(Row))
                className="justify-content-center"
              >
                <strong>
                  <i>
                    vs
                  </i>
                </strong>
              </ForwardRef(Bootstrap(Row))>
              <ForwardRef(Bootstrap(Row))
                className="justify-content-center"
              >
                <p
                  style={
                    Object {
                      "color": "green",
                    }
                  }
                >
                  
                </p>
              </ForwardRef(Bootstrap(Row))>
            </h2>
          </ForwardRef(Bootstrap(Row))>,
          <ForwardRef(Bootstrap(Row))
            className="justify-content-center"
          >
            <h3
              className="mr-2"
              style={
                Object {
                  "color": "white",
                }
              }
            >
              Choose your hand
               
            </h3>
            <h3
              style={
                Object {
                  "color": "red",
                }
              }
            >
              
            </h3>
          </ForwardRef(Bootstrap(Row))>,
          <ForwardRef(Bootstrap(Row))>
            <ChooseHand
              moves={Array []}
              onChoose={[Function]}
              round={1}
              user=""
            />
          </ForwardRef(Bootstrap(Row))>,
          <ShowHandWinner
            handleCloseModal={[Function]}
            round={1}
            show={false}
          />,
          <ShowChampion
            homeScreen={[Function]}
            playAgain={[Function]}
            show={false}
          />,
          ";",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef(Bootstrap(Col))>
              <input
                alt="Home icon"
                id="exit-img"
                onClick={[Function]}
                src="./back.png"
                style={
                  Object {
                    "background": "transparent",
                    "height": "40px",
                    "width": "40px",
                  }
                }
                type="image"
              />
            </ForwardRef(Bootstrap(Col))>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <input
                alt="Home icon"
                id="exit-img"
                onClick={[Function]}
                src="./back.png"
                style={
                  Object {
                    "background": "transparent",
                    "height": "40px",
                    "width": "40px",
                  }
                }
                type="image"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Home icon",
                "id": "exit-img",
                "onClick": [Function],
                "src": "./back.png",
                "style": Object {
                  "background": "transparent",
                  "height": "40px",
                  "width": "40px",
                },
                "type": "image",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": undefined,
              "propTypes": undefined,
              "render": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "propTypes": undefined,
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <h1
              style={
                Object {
                  "color": "white",
                  "fontSize": "4rem",
                }
              }
            >
              Round 
              1
            </h1>,
            "className": "justify-content-center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Round ",
                1,
              ],
              "style": Object {
                "color": "white",
                "fontSize": "4rem",
              },
            },
            "ref": null,
            "rendered": Array [
              "Round ",
              1,
            ],
            "type": "h1",
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "propTypes": undefined,
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <h2
              style={
                Object {
                  "color": "white",
                }
              }
            >
              <ForwardRef(Bootstrap(Row))
                className="justify-content-center"
              >
                <p
                  style={
                    Object {
                      "color": "red",
                    }
                  }
                >
                  
                </p>
              </ForwardRef(Bootstrap(Row))>
              <ForwardRef(Bootstrap(Row))
                className="justify-content-center"
              >
                <strong>
                  <i>
                    vs
                  </i>
                </strong>
              </ForwardRef(Bootstrap(Row))>
              <ForwardRef(Bootstrap(Row))
                className="justify-content-center"
              >
                <p
                  style={
                    Object {
                      "color": "green",
                    }
                  }
                >
                  
                </p>
              </ForwardRef(Bootstrap(Row))>
            </h2>,
            "className": "justify-content-center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <ForwardRef(Bootstrap(Row))
                  className="justify-content-center"
                >
                  <p
                    style={
                      Object {
                        "color": "red",
                      }
                    }
                  >
                    
                  </p>
                </ForwardRef(Bootstrap(Row))>,
                <ForwardRef(Bootstrap(Row))
                  className="justify-content-center"
                >
                  <strong>
                    <i>
                      vs
                    </i>
                  </strong>
                </ForwardRef(Bootstrap(Row))>,
                <ForwardRef(Bootstrap(Row))
                  className="justify-content-center"
                >
                  <p
                    style={
                      Object {
                        "color": "green",
                      }
                    }
                  >
                    
                  </p>
                </ForwardRef(Bootstrap(Row))>,
              ],
              "style": Object {
                "color": "white",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <p
                    style={
                      Object {
                        "color": "red",
                      }
                    }
                  >
                    
                  </p>,
                  "className": "justify-content-center",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "",
                    "style": Object {
                      "color": "red",
                    },
                  },
                  "ref": null,
                  "rendered": "",
                  "type": "p",
                },
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": undefined,
                  "propTypes": undefined,
                  "render": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <strong>
                    <i>
                      vs
                    </i>
                  </strong>,
                  "className": "justify-content-center",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <i>
                      vs
                    </i>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "vs",
                    },
                    "ref": null,
                    "rendered": "vs",
                    "type": "i",
                  },
                  "type": "strong",
                },
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": undefined,
                  "propTypes": undefined,
                  "render": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": <p
                    style={
                      Object {
                        "color": "green",
                      }
                    }
                  >
                    
                  </p>,
                  "className": "justify-content-center",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "",
                    "style": Object {
                      "color": "green",
                    },
                  },
                  "ref": null,
                  "rendered": "",
                  "type": "p",
                },
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": undefined,
                  "propTypes": undefined,
                  "render": [Function],
                },
              },
            ],
            "type": "h2",
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "propTypes": undefined,
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <h3
                className="mr-2"
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                Choose your hand
                 
              </h3>,
              <h3
                style={
                  Object {
                    "color": "red",
                  }
                }
              >
                
              </h3>,
            ],
            "className": "justify-content-center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Choose your hand",
                  " ",
                ],
                "className": "mr-2",
                "style": Object {
                  "color": "white",
                },
              },
              "ref": null,
              "rendered": Array [
                "Choose your hand",
                " ",
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "style": Object {
                  "color": "red",
                },
              },
              "ref": null,
              "rendered": "",
              "type": "h3",
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "propTypes": undefined,
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ChooseHand
              moves={Array []}
              onChoose={[Function]}
              round={1}
              user=""
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "moves": Array [],
              "onChoose": [Function],
              "round": 1,
              "user": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "propTypes": undefined,
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handUser1": undefined,
            "handUser2": undefined,
            "handleCloseModal": [Function],
            "round": 1,
            "show": false,
            "winner": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "homeScreen": [Function],
            "playAgain": [Function],
            "show": false,
            "winner": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        ";",
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "propTypes": undefined,
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
